src/dft.cpp

```diff
@@ -246,12 +246,12 @@ void dft_chunk::update_dft(double time) {
         f[cmp] = w * fc->f[c][cmp][idx];
 
     if (numcmp == 2) {
-      complex<double> fc(f[0], f[1]);
+      complex<realnum> fc(f[0], f[1]);
       for (int i = 0; i < Nomega; ++i)
         dft[Nomega * idx_dft + i] += dft_phase[i] * fc;
     }
     else {
-      double fr = f[0];
+      realnum fr = f[0];
       for (int i = 0; i < Nomega; ++i)
         dft[Nomega * idx_dft + i] += dft_phase[i] * fr;
```

Comment:
On clang, this expression results in subpar code generation with almost no performance increase over double.
I'd recommend to compare the performance of a DFT monitor on the Yee grid with the current implementation and with this alternative:
```
dft[Nomega * idx_dft + i] += std::complex<realnum>{fr * dft_phase[i].real(), fr * dft_phase[i].imag()};
```

Reply:
I am not able to see any significant difference in performance using this alternative update formulation with either gcc or clang. I used a slightly different test (see [gist](https://gist.github.com/oskooi/2279c48206a3594d35c6351f20e08726)) replacing `add_flux` with `add_dft_fields` and `yee_grid=True` as you suggested [above](https://github.com/NanoComp/meep/pull/1675#issuecomment-880950260).

Reply:
I have switched to this modified DFT field update anyway because it yields improved performance for clang and libc++.